src/client/Microsoft.Identity.Client/Cache/CacheSessionManager.cs

```diff
@@ -38,7 +38,14 @@ internal class CacheSessionManager : ICacheSessionManager
         public async Task<MsalAccessTokenCacheItem> FindAccessTokenAsync()
         {
             await RefreshCacheForReadOperationsAsync(CacheEvent.TokenTypes.AT).ConfigureAwait(false);
-            return await TokenCacheInternal.FindAccessTokenAsync(_requestParams).ConfigureAwait(false);
+            MsalAccessTokenCacheItem msalAccessToken = await TokenCacheInternal.FindAccessTokenAsync(_requestParams).ConfigureAwait(false);
+
+            if (msalAccessToken == null && _requestParams.RequestContext.ApiEvent.CacheRefresh == null)
```

Comment:
I don't think this logic belongs here. CacheManager does not know about this. Logic should go to request classes.